rust/azure_iot_operations_protocol/src/rpc_command/executor.rs

```diff
@@ -212,6 +212,42 @@ impl<TResp: PayloadSerialize> ResponseBuilder<TResp> {
         }
         Ok(())
     }
+
+    /// Helper function to add user-defined application error headers to `response`.
+    ///
+    /// `application_error_code` required to be a non-empty `String`.
+    /// `application_error_payload` is optional and can be an empty `String`, in which case it is ignored and not added to `response`.
+    ///
+    /// Returns `Ok()` if `application_error_code` is not an empty `String`.
+    ///
+    /// # Errors
+    /// Returns an Error with the `String` "`application_error_code` cannot be empty" if `application_error_code` is an empty string.
+    pub fn add_application_error_headers(
+        response: &mut Response<TResp>,
+        application_error_code: String,
+        application_error_payload: String,
+    ) -> Result<(), String> {
+        if application_error_code.is_empty() {
```

Comment:
`is_empty` will return false if `application_error_code` is just whitespace

```suggestion
        if application_error_code.trim().is_empty() {
```

Reply:
Done.